x-pack/auditbeat/module/system/package/package.go

```diff
@@ -0,0 +1,537 @@
+		return errors.Wrap(err, "error writing packages to disk")
+	}
+	return nil
+}
+
+func getPackages(osFamily string) (packages []*Package, err error) {
+	switch osFamily {
+	case redhat:
+		// TODO: Implement RPM
+		err = errors.New("RPM not yet supported")
+	case debian:
+		packages, err = listDebPackages()
+		if err != nil {
+			err = errors.Wrap(err, "error getting DEB packages")
+		}
+	case darwin:
+		packages, err = listBrewPackages()
+		if err != nil {
+			err = errors.Wrap(err, "error getting Homebrew packages")
+		}
+	default:
+		err = errors.Errorf("unknown OS %v - this should not have happened", osFamily)
+	}
+
+	return
+}
+
+func listDebPackages() ([]*Package, error) {
+	file, err := os.Open(dpkgStatusFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error opening %s", dpkgStatusFile)
+	}
+	defer file.Close()
+
+	var packages []*Package
+	pkg := &Package{}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(strings.TrimSpace(line)) == 0 {
+			// empty line signals new package
+			packages = append(packages, pkg)
+			pkg = &Package{}
+			continue
+		}
+		if strings.HasPrefix(line, " ") {
+			// not interested in multi-lines for now
+			continue
+		}
+		words := strings.SplitN(line, ":", 2)
+		if len(words) != 2 {
+			return nil, fmt.Errorf("the following line was unexpected (no ':' found): '%s'", line)
+		}
+		value := strings.TrimSpace(words[1])
+		switch strings.ToLower(words[0]) {
+		case "package":
+			pkg.Name = value
+		case "architecture":
+			pkg.Arch = value
+		case "version":
+			pkg.Version = value
+		case "description":
+			pkg.Summary = value
+		case "installed-size":
+			pkg.Size, err = strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return nil, errors.Wrapf(err, "error converting %s to int", value)
+			}
+		default:
+			continue
+		}
+	}
+	if err = scanner.Err(); err != nil {
+		return nil, errors.Wrap(err, "error scanning file")
+	}
+	return packages, nil
+}
+
+func listBrewPackages() ([]*Package, error) {
+	packageDirs, err := ioutil.ReadDir(homebrewCellarPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading directory %s", homebrewCellarPath)
+	}
+
+	var packages []*Package
+	for _, packageDir := range packageDirs {
+		if !packageDir.IsDir() {
+			continue
+		}
+		pkgPath := path.Join(homebrewCellarPath, packageDir.Name())
+		versions, err := ioutil.ReadDir(pkgPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading directory: %s", pkgPath)
+		}
+
+		for _, version := range versions {
+			if !version.IsDir() {
+				continue
+			}
+			pkg := &Package{
+				Name:        packageDir.Name(),
+				Version:     version.Name(),
+				InstallTime: version.ModTime(),
+			}
+
+			// read formula
+			formulaPath := path.Join(homebrewCellarPath, pkg.Name, pkg.Version, ".brew", pkg.Name+".rb")
+			file, err := os.Open(formulaPath)
+			if err != nil {
+				//fmt.Printf("WARNING: Can't get formula for package %s-%s\n", pkg.Name, pkg.Version)
```

Comment:
Log a warning or debug message?

Reply:
Ah yes, good point. We don't have a logger in hand there at this time unfortunately. But I'll open a follow-up PR to improve Homebrew package collection that is going to address this as well. I have the code mostly ready.